kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -96,6 +96,20 @@ def check(self, instance):
         for job_tags, job_count in iteritems(self.job_failed_count):
             self.monotonic_count(scraper_config['namespace'] + '.job.failed', job_count, list(job_tags))
 
+    def _filter_metric(self, metric, scraper_config):
+        # name is like "kube_pod_execution_duration"
+        name_part = metric.name.split("_", 3)
+        family = name_part[1]
+        tags = ["name:" + family]
+        for sample in metric.samples:
+            if "namespace" in sample[1]:
+                ns = sample[1]["namespace"]
+                tags.append("kube_namespace:" + ns)
+                break
+        self.count(scraper_config['namespace'] + ".telemetry.collector.metrics.count", len(metric.samples), tags=tags)
```

Comment:
Are we just using `_filter_metric` for the side effect of sending a count (`self.count()`) ? 

If so, would overriding `_send_telemetry_counter` work for this case (semantically more correct IMHO) ?

Example:

```python

    def _send_telemetry_counter(self, metric_name, val, scraper_config):
        super(KubernetesState, self)._send_telemetry_counter(metric_name, val, scraper_config)

        # [...]
        
        self.count(scraper_config['namespace'] + ".telemetry.collector.metrics.count", len(metric.samples), tags=tags)

```

Reply:
`_filter_metric ` is call on all input metrics, `_send_telemetry_counter` only on selected metrics. So it will not have the same effect.
But thanks for your remark: I'm updated the code in order to use `_send_telemetry_counter` from the `_filter_metric`